lib/full-site-editing/class-wp-rest-templates-controller.php

```diff
@@ -199,6 +203,11 @@ public function update_item( $request ) {
 			return new WP_Error( 'rest_template_not_found', __( 'No templates exist with that id.', 'gutenberg' ), array( 'status' => 404 ) );
 		}
 
+		if ( isset( $request['reverted_file_content'] ) && isset( $request['content'] ) && $request['reverted_file_content'] === $request['content'] ) {
```

Comment:
I wonder if there is any way to get consistent markup out of `$template->content`? I wanted to compare the content with the file template's content but looks like the JS serializer gives a different markup from the PHP serializer.

Reply:
> Couldn't we just rely on isset( $request['is_custom'] ) && 'false' === $request['is_custom'] to determine that this is a revert call?

I think we need a way to differentiate between 1 - retrieving the non-custom content and 2 - reverting to that.  Reverting based on presence of `$request['is_custom'] === false` means we cannot access that non-custom content without reverting the post as well.  🤔

Reply:
Maybe another request param like `is_revert` could get around needing to compare content? 🤔

Reply:
It's related to saving a template for the first time. It would save as `is_custom = false` since we aren't updating its state on the client side. So we rely on the content. To get around this, we would need to set `is_custom` to true when we start making changes to the template.

Reply:
@Addison-Stavlo It's not super clear by the diff, but I was specifically talking about `update_item`; you can still `get_item` a non-custom item.
(Unless I've misunderstood your comment? 🤔 )

@david-szabo97 Apologies if these are dumb comments, but I really forgot _everything_ I researched while working on this issue. 😓 
When we save a template for the first time (and subsequent times as well, I think?), we only really send ID and content.

<img width="592" alt="Screenshot 2021-03-31 at 10 57 35" src="https://user-images.githubusercontent.com/2070010/113127258-754c5180-9210-11eb-8ee9-bb08333f4b2b.png">

`$request['is_custom']` is undefined, and the `isset( $request['is_custom'] ) && 'false' === $request['is_custom']` check will fail.

Reply:
There is also an issue I encountered while trying to use the implementation you suggest. For some reason the selector that takes care of collecting the changed properties of the entity, doesn't return the `is_custom` property as changed. It only happens with boolean properties. Changing `true`/`false` to `'yes'`/`'no'` gets around this problem.

According to our discussion, I renamed `is_custom` to `source` and it uses two string values: `theme` and `custom`.